include/deal.II/base/big_mpi_compat.h

```diff
@@ -0,0 +1,331 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+/**
+ *
+ * BigMPICompat - a tiny MPI 4.x compatibility library
```

Comment:
Could we handle this like the https://github.com/dealii/dealii/tree/master/include/deal.II/base/std_cxx17 and https://github.com/dealii/dealii/tree/master/include/deal.II/base/std_cxx20? I guess there will be a time when we can remove this file?

Reply:
It can be removed once we require MPI 4.0. As they are barely releasing this year (OpenMPI is still on 3.1), I would expect this to take 10 years or so. :-)